packages/financial-templates-lib/src/logger/Logger.js

```diff
@@ -42,11 +42,25 @@ async function waitForLogger(logger) {
 // If the log entry contains an error then extract the stack trace as the error message.
 function errorStackTracerFormatter(logEntry) {
   if (logEntry.error) {
-    logEntry.error = logEntry.error.stack;
+    logEntry.error = handleRecursiveErrorArray(logEntry.error);
   }
   return logEntry;
 }
 
+// Handle case where `error` is an array of errors and we want to display all
+// of the error stacks recursively. i.e. `error` is in the shape:
+// [[Error, Error], [Error], [Error, Error]]
+function handleRecursiveErrorArray(error) {
```

Comment:
This prints out recursive errors to Slack like so:

```
[warn] nick-local-testing (SyntheticPegMonitor)⭢Unable to get volatility data, missing historical price data
• error:
- 0: ["Error: CryptoWatch-uniswap-v2-fraxusdc: undefined lastUpdateTime\n    at CryptoWatchPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:71:13)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26\n    at Array.map (<anonymous>)\n    at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:118:26\n    at Array.map (<anonymous>)\n    at BasketSpreadPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:116:58)\n    at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)\n    at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)\n    at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)\n    at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)\n    at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)"]
2:06
- 1: ["Error: CryptoWatch-uniswap-v2-esdusdc: undefined lastUpdateTime\n    at CryptoWatchPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:71:13)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26\n    at Array.map (<anonymous>)\n    at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:118:26\n    at Array.map (<anonymous>)\n    at BasketSpreadPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:116:58)\n    at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)\n    at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)\n    at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)\n    at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)\n    at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)"]
2:06
- 2: ["Error: Uniswap-0xd4405f0704621dbe9d4dea60e128e0c3b26bddbd time 1612289196 is earlier than TWAP window\n    at UniswapPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/UniswapPriceFeed.js:68:13)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26\n    at Array.map (<anonymous>)\n    at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:118:26\n    at Array.map (<anonymous>)\n    at BasketSpreadPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:116:58)\n    at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)\n    at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)\n    at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)\n    at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)\n    at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)"]
2:06
- 3: [["Error: CryptoWatch-bitfinex-usdtusd: undefined lastUpdateTime\n    at CryptoWatchPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:71:13)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26\n    at Array.map (<anonymous>)\n    at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26\n    at Array.map (<anonymous>)\n    at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:125:26\n    at Array.map (<anonymous>)\n    at BasketSpreadPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:123:50)\n    at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)\n    at SyntheticPegMonitor-MESSAGE REDACTED DUE TO LENGTH-ceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26\n    at Array.map (<anonymous>)\n    at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)\n    at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:125:26\n    at Array.map (<anonymous>)\n    at BasketSpreadPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/BasketSpreadPriceFeed.js:123:50)\n    at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)\n    at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)\n    at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)\n    at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)\n    at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)"]]
• lookback: 3600
2:06
Something went wrong in the winston formatter!
Error:TypeError: Cannot read property 'length' of undefined
logInfo:{"at":"SyntheticPegMonitor","message":"Unable to get volatility data, missing historical price data","pricefeed":"Medianizer","errorInformation":{"latestTime":1612292811},"lookback":3600,"level":"warn","bot-identifier":"nick-local-testing"}
2:07
Something went wrong in the winston formatter!
Error:TypeError: Cannot read property 'length' of undefined
logInfo:{"at":"SyntheticPegMonitor","message":"Unable to get volatility data, missing historical price data","pricefeed":"Medianizer","errorInformation":{"latestTime":1612292834},"lookback":3600,"level":"warn","bot-identifier":"nick-local-testing"}
2:08
[warn] nick-local-testing (SyntheticPegMonitor)⭢Unable to get volatility data, missing historical price data
• pricefeed: Medianizer
• error: null
• lookback: 3600
2:11
[warn] nick-local-testing (SyntheticPegMonitor)⭢Unable to get volatility data, missing historical price data
• error:
- 0: Error: CryptoWatch-coinbase-pro-btcusd: undefined lastUpdateTime
   at CryptoWatchPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:71:13)
   at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26
   at Array.map (<anonymous>)
   at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)
   at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)
   at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)
   at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)
   at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)
   at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)
   at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)
- 1: Error: CryptoWatch-binance-btcusdt: undefined lastUpdateTime
   at CryptoWatchPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:71:13)
   at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26
   at Array.map (<anonymous>)
   at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)
   at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)
   at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)
   at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)
   at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)
   at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)
   at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)
2:11
- 2: Error: CryptoWatch-bitstamp-btcusd: undefined lastUpdateTime
   at CryptoWatchPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:71:13)
   at /Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:44:26
   at Array.map (<anonymous>)
   at MedianizerPriceFeed.getHistoricalPrice (/Users/nicholaspai/UMA/protocol/packages/financial-templates-lib/src/price-feed/MedianizerPriceFeed.js:42:44)
   at SyntheticPegMonitor._calculateHistoricalVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:337:28)
   at SyntheticPegMonitor._checkPricefeedVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:293:24)
   at SyntheticPegMonitor.checkPegVolatility (/Users/nicholaspai/UMA/protocol/packages/monitors/src/SyntheticPegMonitor.js:186:28)
   at retry.retries (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:254:33)
   at run (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:229:7)
   at Poll (/Users/nicholaspai/UMA/protocol/packages/monitors/index.js:361:5)
• lookback: 3600
```

Reply:
(I forced an error to be thrown in the constituent price feeds to simulate how errors would be logged)